### PR TITLE
[CDAP-20229]: Add new config for enabling/disabling git integration

### PR DESCRIPTION
**Pull request by samdgupi**
Added a new config for controlling git integration.

Testing: Not needed

### CODE REVIEW

File: cdap-common/src/main/java/io/cdap/cdap/common/conf/Constants.java [hunk 1]
```diff
@@ -2063,4 +2063,11 @@ public static final class MetadataConsumer {
     public static final String METADATA_CONSUMER_EXTENSIONS_ENABLED_LIST = "metadata.consumer.extensions.enabled.list";
     public static final String METADATA_CONSUMER_EXTENSIONS_DIR = "metadata.consumer.extensions.dir";
   }
+
+  /**
+   * Configuration for source control management
+   */
+  public static final class SourceControl {
+    public static final String GIT_INTEGRATION_ENABLED = "scm.git.enable";
```
Comment (sumengwang) [inline cdap-common/src/main/java/io/cdap/cdap/common/conf/Constants.java]: My suggestion is 'feature.source.control.management.git.enabled'

Comment (samdgupi) [reply cdap-common/src/main/java/io/cdap/cdap/common/conf/Constants.java]: I also see we use Feature enum for adding feature flags. What is the recommendation here vis-a-vis config vs feature flag ?
cc: @arjan-bal @de-lan

File: cdap-common/src/main/resources/cdap-default.xml [hunk 2]
```diff
@@ -5593,4 +5593,15 @@
       Health Check HTTP service bind port; if 0, binds to a random port
     </description>
   </property>
+
+  <!-- Source Control Configuration  -->
```
Comment (de-lan) [inline cdap-common/src/main/resources/cdap-default.xml]: suggestion: `Source Control Configuration` --> `Configuration for Source Control Management`

File: cdap-common/src/main/resources/cdap-default.xml [hunk 3]
```diff
@@ -5593,4 +5593,15 @@
...
+  <property>
+    <name>feature.source.control.management.git.enabled</name>
+    <value>false</value>
+    <final>true</final>
+    <description>
+      If true, source control using git will be enabled.
```
Comment (de-lan) [inline cdap-common/src/main/resources/cdap-default.xml]: suggestion, maybe use: `source control management`?

**Review by de-lan — APPROVED** [review]
LGTM, just some nits

**Comment by de-lan** [comment]
Also pls squash the commits to keep it clean

File: cdap-common/src/main/resources/cdap-default.xml (hunk 2 above)
Comment (samdgupi) [reply cdap-common/src/main/resources/cdap-default.xml]: ✅

File: cdap-common/src/main/resources/cdap-default.xml (hunk 3 above)
Comment (samdgupi) [reply cdap-common/src/main/resources/cdap-default.xml]: ✅

**Comment by samdgupi** [comment]
> 

The squash and merge option seems blocked for me

**Comment by de-lan** [comment]
> > 
> 
> The squash and merge option seems blocked for me

Yeah it's blocked for the repo. We have to manually squash it locally and force push to this branch

**Pull request merged**